src/Aspire.Hosting.AWS/Provisioning/AWSProvisionerExtensions.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.AWS;
+using Aspire.Hosting.AWS.CDK;
+using Aspire.Hosting.AWS.CloudFormation;
+using Aspire.Hosting.AWS.Provisioning;
+using Aspire.Hosting.Lifecycle;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding support for generating AWS resources dynamically during application startup.
+/// </summary>
+public static class AWSProvisionerExtensions
```

Comment:
Is there reason this needs to be public or can we start with it being `internal`?